spec/models/user_spec.rb

```diff
@@ -1,6 +1,29 @@
 require 'spec_helper'
 
 describe User do
+  describe '#accessible_organizations' do
+    context 'when one organizations' do
+      let(:user) { create(:user,
+                          metadata: Mumukit::Auth::Metadata.new( atheneum: {permissions: 'pdep/*'}))}
+      it { expect(user.accessible_organizations.size).to be 1}
+    end
+    context 'when two organizations' do
+      let(:user) { create(:user,
+                          metadata: Mumukit::Auth::Metadata.new( atheneum: {permissions: 'pdep/*:alcal/*'}))}
+      it { expect(user.accessible_organizations.size).to be 2 }
+    end
+    context 'when all grant present organizations' do
+      let(:user) { create(:user,
+                          metadata: Mumukit::Auth::Metadata.new( atheneum: {permissions: 'pdep/*:*'}))}
+      it { expect(user.accessible_organizations.size).to be 1 }
+    end
+    context 'when one organization appears twice' do
+      let(:user) { create(:user,
+                          metadata: Mumukit::Auth::Metadata.new( atheneum: {permissions: 'pdep/*:pdep/*'}))}
+      it { expect(user.accessible_organizations.size).to be 1 }
+    end
+  end
+
```

Comment:
Could we also add some tests to the new controller methods?

I understand that we can not test the whole cycle, since there is JS involved, but we could at least tests in an integration test that the JS code is rendered when it has to.